### ISSUE TITLE
Mattermost not working with Greenlight

### ISSUE THREAD

**Issue by Yumash** [blindsidenetworks/mattermost-plugin-bigbluebutton#66]
It works perfect with single BBB installation, but when Greenlight was installed, there came redirects from URL like

https://bbb/bigbluebutton/api/join?fullName=NAME&meetingID=ID&password=PASS&joinViaHtml5=true&checksum=CHECKSUM

to URL like 

https://bbb/b

**Comment by ffdixon** [comment]
Hmm ... installing GreenLight shouldn't affect other API integrations.  The key file that Greenlight installs is 
~~~bash
$ cat /etc/bigbluebutton/nginx/greenlight-redirect.nginx
location = / {
  return 307 /b;
}
~~~

If you move that file and restart nginx, does Mattermost start working again?

**Comment by Yumash** [comment]
nope, but i have found the way to make it work. I have disabled Greenlight auth, enabled LDAP, and everything magically started work as supposed. I think that it would be great to have more expected behaviour for integration or just a better documentation, because i suppose that there are permissions issues somewhere

**Comment by cadavre** [comment]
How is Mattermost integration affecting this? Greenlight is not what BBB plugin for Mattermost is launching. Or am I missing smth?

**Comment by Yumash** [comment]
@cadavre i suppose that if Greenlight is a part (supported part) of BBB, the installation of it should not break the integration MM<->BBB.

The case was that after i installed Greenlight, my integration became broken, that's all. It seems we are missing some documentation for expected behaviour. I think there's permissions(authorization) issues, because Greenlight puts some security to BBB as is.

**Comment by cadavre** [comment]
Roger @Yumash . The case is I'm using Greenlight too and it didn't broke a thing in MM<->BBB integration via this plugin. Looks like you have some custom config.

Did you used bbb-install.sh and Greenlight via Docker?

**Comment by Yumash** [comment]
@cadavre all was from the docs. I installed both BBB+Greenlight with bbb-install.sh command. 

But before it i have installed only BBB (with bbb-install) - the integration worked perfectly.

After i have installed BBB+GL, integration broked till the special magic i did in https://github.com/blindsidenetworks/mattermost-plugin-bigbluebutton/issues/66#issuecomment-553580322

**Comment by eyebank** [comment]
So @Yumash I noticed the same thing that when greenlight is installed and i launch BBB with the plug in, the url for greenlight appears. Like: "https://mydomain.org/b" . i'm gonna try to disable greenlight (your method) to see if BBB launches. snapshot first!

**Comment by rottaran** [comment]
This was me driving almost insane :) The issue is, that the location "/b" in nginx matches everything **beginning** with b, which can be even more than just /bigbluebutton and /bbb. 
I solved it this way:

```
location ~ (^/b$|^/b/) {
      proxy_pass          http://10.88.0.40:80;
      proxy_set_header    Host              $host;
      proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
      proxy_set_header    X-Forwarded-Proto $scheme;
      proxy_http_version  1.1;
}
```

**Comment by eyebank** [comment]
So, I resolved my issue. Chrome broke things. I migrated to firefox and can use both front ends BBB Plugin and Greenlight) without issue.

**Comment by Joshndroid** [comment]
Interesting... Apologies to bring up an old thread but i had set up bigbluebutton from the docs and have the mattermost plugin working... i then thought i would get rid of the demo stuff and install greenlight... once it did that the plugin broke, giving me 404 errors. I tried removing the proxy auto route to /b, which didn't work. I tried further combinations within the plugin; adding the /b to URL, removing it, changing browsers, etc to no avail. Once i put demo back in, removed greenlight, the plugin works and takes be straight to the room created by mattermost. While im not to concerned as i can live with it in this state as its only for home network use, i found this through googling and wish to express i did see this issue. Greenlight was also installed exactly as the docks entail, much like it did for bbb on ubuntu server 18.04. cheers.
